filebeat/input/file/state.go

```diff
@@ -10,24 +10,26 @@ import (
 
 // State is used to communicate the reading state of a file
 type State struct {
-	Source      string      `json:"source"`
-	Offset      int64       `json:"offset"`
-	Finished    bool        `json:"-"` // harvester state
-	Fileinfo    os.FileInfo `json:"-"` // the file info
-	FileStateOS StateOS
+	Finished    bool          `json:"-"` // harvester state
+	Fileinfo    os.FileInfo   `json:"-"` // the file info
+	Source      string        `json:"source"`
+	Offset      int64         `json:"offset"`
 	Timestamp   time.Time     `json:"timestamp"`
 	TTL         time.Duration `json:"ttl"`
+	Type        string        `json:"type"`
```

Comment:
I'm hoping to have only one type in the future. More accurate here would be `ProspectorType` but as I try to keep the state as "independent" as possible with potentially have it in libbeat one day, I would just call it `Type`.